sql/core/src/test/resources/sql-tests/inputs/group-by-all.sql

```diff
@@ -0,0 +1,71 @@
+-- group by all
+-- see https://www.linkedin.com/posts/mosha_duckdb-firebolt-snowflake-activity-7009615821006131200-VQ0o
+
+create temporary view data as select * from values
+  ("USA", "San Francisco", "Reynold", 1, 11.0),
+  ("USA", "San Francisco", "Matei", 2, 12.0),
+  ("USA", "Berkeley", "Xiao", 3, 13.0),
+  ("China", "Hangzhou", "Wenchen", 4, 14.0),
+  ("China", "Shanghai", "Shanghaiese", 5, 15.0),
+  ("Korea", "Seoul", "Hyukjin", 6, 16.0),
+  ("UK", "London", "Sean", 7, 17.0)
+  as data(country, city, name, id, power);
+
+-- basic
+select country, count(*) from data group by ALL;
+
+-- different case
+select country, count(*) from data group by aLl;
+
+-- a column named "all" would still work
+select all, city, count(*) from (select country as all, city, id from data) group by all, city;
+
+-- a column named "all" should take precedence over the normal group by all expansion
+-- if all refers to the column, then the following should return 3 rows.
+-- if all refers to the global aggregate, then 1 row.
+SELECT count(1) FROM VALUES(1), (2), (3) AS T(all) GROUP BY all;
```

Comment:
@cloud-fan i think this case happens to work today because ResolveGroupByAll runs after ResolveReferences. But since they are in the same batch, and ResolveReference doesn't run until fix point, is it possible that this won't work for more complex cases?

Reply:
Shall we merge the rule `ResolveGroupByAll` into `ResolveReference` to avoid potential issues? 
@cloud-fan is working on this for existing rules: https://github.com/apache/spark/pull/38888

Reply:
I don't think we should do an one off merging of some resolution logic into ResolveReference. If we want to do it, we should do it together in one shot. (It's also not clear to me if we should actually merge all resolution together.)

Reply:
I think it's better to put column resolution in one rule. Relying on the rule order is not robust. Let's look at an example: let's say the plan tree is `A -> B -> C`, and only the leaf node `A` is resolved for now.
1. `ResolveReference` resolves columns on `B`, but `B` is not fully resolved as it contains an unresolved function
2. `ResolveFunctions` resolves functions in `B`, now `B` is fully resolved
3. `ResolveGroupBy` resolves column `ALL` in `C`. This breaks the expected resolution order.

Ideally, we want to run different column resolution rules in order for each plan node, not the entire plan tree. Merging them into a single rule is the most straightforward approach, or we need to design a complicated framework to achieve this.

Reply:
Can we take that discussion elsewhere? This is about group by all, not fixing the analyzer tech debt.

Reply:
OK, to make `ResolveGroupByAll` a bit more robust, we can do
```
object ResolveGroupByAll extends Rule[LogicalPlan] {
  def apply(plan: LogicalPlan) = ResolveReference(plan).transform ...
}
```

Reply:
addressed as discussed offline, by adding another resolved check on child.